providers/printers.rb

```diff
@@ -118,18 +114,9 @@ def delete_printer(prt_name)
 
         curr_ptr_name   = printer.name.gsub(" ","+")
         curr_ptr_id     = printer.manufacturer.gsub(" ","-") + "-" + printer.model.gsub(" ","_")
-        gecos_ptr_name  = printer.name.gsub(" ","+")
-
-		if `/usr/bin/lpoptions -p #{gecos_ptr_name}`.length <= 1
-            install_printer(curr_ptr_name, curr_ptr_id, printer.model, printer.uri, oppolicy)
-        else
-            cups_list.each do |cups_printer|
-                if cups_printer.eql? gecos_ptr_name
-                    update_printer(curr_ptr_name, oppolicy)
-                    break
-                end
-            end
-        end
+        create_ppd(curr_ptr_name, printer.model, curr_ptr_id)
+        install_or_update_printer(curr_ptr_name, printer.uri, printer.oppolicy)
```

Comment:
As in previous line. A performance check is needed before accepting this behaviour (update on every agent run).